### ISSUE TITLE
Intersection and Union support

### ISSUE THREAD

**Issue by sstenchlak** [mff-uk/dataspecer#668]
There might be a need for new semantic entities roughly corresponding to owl:unionOf and owl:intersectionOf in CME and the rest of Dataspecer. A simple example is "A Person may have a Cat and a Dog as a Pet" which consists of three classes (note that there is no Pet class), the union that is linked to Dog and Cat, and one relation "pet" from Person to the union.

This is not a requirement yet!

```mermaid
graph LR
    A[Person class] -- pet relation --> B((∪))
    B --> C[Dog class]
    B --> D[Cat class]
```

**Comment by skodapetr** [comment]
@sstenchlak  Any idea how is this represented? Do we have names for the connections like source, targets  ? 
I presume there can be multiple targets?

**Comment by jakubklimek** [comment]
@skodapetr You mean from the dataspecer data model point of view, right? Because otherwise, this corresponds to a class defined as `owl:unionOf <C1>, <C2>, <C3>` or `owl:intersectionOf <C1>, <C2>, <C3>`.

**Comment by skodapetr** [comment]
Yes, so it is basically a generalization of the relationship for the editor. 
We have BinaryRelationship with source, target and waypoints. 
This would be Relationship with sourceS, targetS, center and waypoints for each source and target.

Is this true?

**Comment by jakubklimek** [comment]
I don't think so... even the owl interpretation is that it is a class. So we still have a binary relationship to the union/intersection classs. What will be new will be how to say what is inside that union/intersection. @sstenchlak already represents this somehow in the Dataspecer model (at least I think so). In the visual model, it will be a new type of relationship between the union/intersection class and its member classes.

**Comment by skodapetr** [comment]
So what you propose is to add a UnionClass and IntersectionClass ? 
But we already have a Class with IS-A (generalization). 
How should those be related?

**Comment by jakubklimek** [comment]
And ComplementOfClass - and the relation will be a new one (something like "member"), not related to generalization

**Comment by skodapetr** [comment]
Each of those three classes (Union, Intersection, Compolement) and the relations will have a different actions / toolbar / menu? Or we just use general: detail, edit, profile, hide, delete ?

**Comment by jakubklimek** [comment]
they can have e.g. "Add to union" etc. actions on top.
